lib/TH/THMemoryFile.c

```diff
@@ -177,7 +177,7 @@ static int THMemoryFile_mode(const char *mode, int *isReadable, int *isWritable)
       size_t i;                                                           \
       for(i = 0; i < n; i++)                                            \
       {                                                                 \
-        ssize_t nByteWritten;                                           \
+        int nByteWritten;                                               \
```

Comment:
This was the case before the change: 'ssize_t' is often defined as 'int' while 'size_t' can be 'unsigned long int' in the same header. The 'ssize_t' is not defined in the C++ standard, so it's not easy to reason about its size. I could use the 'intmax_t' type, but it might be slower.